cryptoki-sys/oasis-headers/pkcs11.h

```diff
@@ -181,6 +181,12 @@ extern "C" {
  * #endif
  */
 
+/* Recommended Unix definitions from above */
+#define CK_PTR *
+#define CK_DECLARE_FUNCTION(returnType, name) returnType name
+#define CK_DECLARE_FUNCTION_POINTER(returnType, name) returnType (* name)
+#define CK_CALLBACK_FUNCTION(returnType, name) returnType (* name)
+
```

Comment:
My recommendation for this is to add a new header file (something like `pkcs11-rust.h`) that defines these rather than doing them in-line.

The following is copied out of a header file I've used to define cryptoki types that has worked on Linux, MacOS and Windows
```C
#pragma once

#ifdef _MSC_VER
#pragma pack(push, cryptoki, 1)
#endif

#if defined _WIN32
#if defined P11_STATIC
#define P11_EXPORT
#elif defined DLL_EXPORT
#define P11_EXPORT    __declspec(dllexport)
#else
#define P11_EXPORT    __declspec(dllimport)
#endif
#else
#define P11_EXPORT
#endif

// Define UNIX defaults for PKCS11 interface

/// 1. CK_PTR: The indirection string for making a pointer to an object.
#define CK_PTR    *

/// 2. CK_DEFINE_FUNCTION(returnType, name): A macro which makes
/// an exportable Cryptoki library function definition out of a
/// return type and a function name.
#define CK_DEFINE_FUNCTION(returnType, name) \
    returnType name

/// 3. CK_DECLARE_FUNCTION(returnType, name): A macro which makes
/// an importable Cryptoki library function declaration out of a
/// return type and a function name.
#define CK_DECLARE_FUNCTION(returnType, name) \
    returnType P11_EXPORT name

/// 4. CK_DECLARE_FUNCTION_POINTER(returnType, name): A macro
/// which makes a Cryptoki API function pointer declaration or
/// function pointer type declaration out of a return type and a
/// function name.
#define CK_DECLARE_FUNCTION_POINTER(returnType, name) \
    returnType P11_EXPORT(*name)

/// 5. CK_CALLBACK_FUNCTION(returnType, name): A macro which makes
/// a function pointer type for an application callback out of
/// a return type for the callback and a name for the callback.
#define CK_CALLBACK_FUNCTION(returnType, name) \
    returnType(*name)

/// 6. NULL_PTR: This macro is the value of a NULL pointer.
#ifndef NULL_PTR
#define NULL_PTR    0
#endif

#include <pkcs11.h>

#ifdef _MSC_VER
#pragma pack(pop, cryptoki)
#endif
```

Reply:
+1. I don't have an environment to test for anything like platform-specific preprocessor stuff, so I opted to make it as small and simple as possible. I'll be sure to steal this snippet when I start committing again.